src/main/java/zpi/algospace/controller/SolutionController.java

```diff
@@ -27,7 +30,7 @@ public class SolutionController {
     public Boolean judgeSolution(@RequestBody SolutionDTO solution) {
 
         try {
-            return solutionService.judgeSolution(solution.toSolution());
+            return solutionService.judgeSolution(solution.toSolution(taskRepository, userRepository));
```

Comment:
wydzielmy to mapowanie do SolutionService, bo nie podoba mi się, że podajemy repozytoria jako argumenty
jeśli już chciałbyś ich użyć to tam to wpierdol( to tez nie jest dobre rozwiazanie, ale wtedy ustawiasz tego beana jako singleton i imho lepiej bedzie) na pewno nie w argumentach